src/query/api/v1/handler/prom/read.go

```diff
@@ -84,14 +87,67 @@ var (
 				params.Now)
 		}
 	}
+
+	// Ratio of queries we make a check for
+	DefaultCheckSampleRate float64 = 0.0
+	// Threshold in % to determine if there's difference in results (1 means 1% diff)
+	DefaultComparePercentThreshold float64 = 1.0
 )
 
+// Compares results a, b to the specified percent threshold
+// Results should be vectors
+func compareResults(a, b *promql.Result, threshold float64) bool {
```

Comment:
What may cause the value different?

Reply:
I think there can be some slight floating point precision errors especially with comparison, so I thought a % threshold would be best